File: examples/stock/history/node/getHistoricData.js
import * as stock from "m3o/stock";

// Get the historic open-close for a given day
async function GetHistoricData() {
  let stockService = new stock.StockService(process.env.MICRO_API_TOKEN);
  let rsp = await stockService.history({
    date: "2020-10-01",
    stock: "AAPL",
  });
  console.log(rsp);
}

await GetHistoricData();

File: examples/stock/history/go/getHistoricData.go
package example

import (
	"fmt"
	"github.com/micro/services/clients/go/stock"
	"os"
)

// Get the historic open-close for a given day
func GetHistoricData() {
	stockService := stock.NewStockService(os.Getenv("MICRO_API_TOKEN"))
	rsp, err := stockService.History(&stock.HistoryRequest{
		Date:  "2020-10-01",
		Stock: "AAPL",
	})
	fmt.Println(rsp, err)
}
